fiftyone/zoo/datasets/__init__.py

```diff
@@ -23,18 +23,72 @@
 logger = logging.getLogger(__name__)
 
 
-def list_zoo_datasets():
+def list_zoo_datasets(tags=None, source=None):
     """Returns the list of available datasets in the FiftyOne Dataset Zoo.
 
+    Example usage::
+
+        import fiftyone as fo
+        import fiftyone.zoo as foz
+
+        #
+        # List all zoo datasets
+        #
+
+        names = foz.list_zoo_datasets()
+        print(names)
+
+        #
+        # List all zoo datasets with (both of) the specified tags
+        #
+
+        names = foz.list_zoo_datasets(tags=["image", "detection"])
+        print(names)
+
+        #
+        # List all zoo datasets available via the given source
+        #
+
+        names = foz.list_zoo_datasets(source="torch")
+        print(names)
+
+    Args:
+        tags (None): only include datasets that have the specified tag or list
+            of tags
+        source (None): only include datasets available via the given source or
+            list of sources
+
     Returns:
-        a list of dataset names
+        a sorted list of dataset names
     """
-    datasets = set()
+    if etau.is_str(source):
+        sources = [source]
+    elif source is not None:
+        sources = list(sources)
+    else:
+        sources, _ = _get_zoo_dataset_sources()
+
```

Comment:
It's important to rely on `_get_zoo_datasets()` to grab the available zoo datasets, because that method contains additional logic that respects the fact that users can [register additional zoo datasets](https://voxel51.com/docs/fiftyone/user_guide/dataset_zoo/api.html#adding-datasets-to-the-zoo).

Reply:
This makes sense. Thanks for adding this